src/validation.h

```diff
@@ -657,6 +658,46 @@ bool InvalidateBlock(CValidationState& state, const CChainParams& chainparams, C
 /** Remove invalidity status from a block and its descendants. */
 void ResetBlockFailureFlags(CBlockIndex* pindex) EXCLUSIVE_LOCKS_REQUIRED(cs_main);
 
+/**
+ * Serialized version of a UTXO set from which an assumed-valid CChainState
+ * can be constructed.
+ */
+class SnapshotMetadata
+{
+public:
+    CDiskBlockIndex m_base_blockheader;
+    uint256 m_utxo_contents_hash = {};
+    uint64_t m_coins_count = 0;
+    /**
+     * Necessary to "fake" the base nChainTx so that we can estimate progress during
+     * snapshot IBD.
+     */
+    unsigned int m_nchaintx = 0;
+    bool m_validation_complete = false;
```

Comment:
In commit "validation: add unused SnapshotMetadata class" (50d8e7c910edec10dcf08beec577c3efd67f2276)

Would add comments to describe these other fields.

In particular, I'm curious why `m_validation_complete` would be a field that gets serialized as opposed to a memory-only state.

Reply:
I'll add comments, but to answer your immediate question: once we complete validation of a snapshot, we update this value to true and write it out to a metadata file on disk. This is so that when we are initializing after a subsequent restart, we know that the snapshot we load in has been validated and we don't need to continue to do a background validation. This would be unnecessary if we renamed the `chainstate_xxx...x` leveldb directory used by the snapshot to `chainstate` after we complete background validation of the snapshot, which is maybe what we should do anyway.